FWCore/Framework/interface/ModuleRegistry.h

```diff
@@ -39,6 +40,8 @@ namespace edm {
 
   class ModuleRegistry {
   public:
+    ModuleRegistry() = default;
+    explicit ModuleRegistry(std::unique_ptr<ModuleTypeResolverBase>);
```

Comment:
It seems to me that this constructor is not implemented yet. Was that intentional?

How do you foresee the information of the resolver type would lead to an argument object? Something along another plugin factory for those + a configuration parameter (e.g. in the top-level PSet) perhaps?